internal/core/command/device.go

```diff
@@ -79,33 +80,17 @@ func commandByNames(dn string, cn string, body string, isPutCommand bool, ctx co
 			return err.Error(), http.StatusInternalServerError
 		}
 	}
-	//once command service has its own persistence layer, we will directly read from the db getCommandByNameAndDeviceID
-	var commands []contract.Command
-	if commands, err = cc.CommandsForDeviceId(d.Id, ctx); err != nil {
+	var command contract.Command
+	command, err = dbClient.GetCommandByNameAndDeviceId(cn, d.Id)
```

Comment:
Why not just `command, err := dbClient.GetCommandByNameAndDeviceId(cn, d.Id)` without the `var` line?

Also, what do you think of turning the first line in this function into a DBClient call?
`d, err := mdc.DeviceForName(dn, ctx)`

Reply:
Absolutely agree with your first question. 
About the second one, I am not sure what you mean , guess it is connected with 

> Should we replace the above client with additional DBClient functions?